ert_shared/storage/extraction_api.py

```diff
@@ -0,0 +1,276 @@
+    for key, response in responses.items():
+        indexes_df = blob_api.add_blob(response.index.to_list())
+        blob_api.flush()
+        response_definition = rdb_api.add_response_definition(
+            name=key, indexes_ref=indexes_df.id, ensemble_name=ensemble_name,
+        )
+        for realization_index, values in response.iteritems():
+            values_df = blob_api.add_blob(values.to_list())
+            blob_api.flush()
+            rdb_api.add_response(
+                name=response_definition.name,
+                values_ref=values_df.id,
+                realization_index=realization_index,
+                ensemble_name=ensemble_name,
+            )
+
+
+def _extract_active_observations(facade):
+    update_step = facade.get_update_step()
+    ministep = update_step[len(update_step) - 1]
+    obs_data = ministep.get_obs_data()
+    if obs_data is None:
+        return None
+    active_observations = dict()
+    for block_num in range(obs_data.get_num_blocks()):
+        block = obs_data.get_block(block_num)
+        obs_key = block.get_obs_key()
+        active_list = [block.is_active(i) for i in range(len(block))]
+        active_observations[obs_key] = active_list
+    return active_observations
+
+
+def _extract_and_dump_update_data(ensemble_id, ensemble_name, rdb_api, blob_api):
+    facade = ERT.enkf_facade
+
+    fs = facade.get_current_fs()
+    realizations = MisfitCollector.createActiveList(ERT.ert, fs)
+
+    active_observations = _extract_active_observations(facade)
+
+    for obs_vector in facade.get_observations():
+        observation_key = obs_vector.getObservationKey()
+        response_key = obs_vector.getDataKey()
+        response_definition = rdb_api._get_response_definition(
+            response_key, ensemble_id
+        )
+
+        if active_observations is not None:
+            active_blob = blob_api.add_blob(active_observations[observation_key])
+            blob_api.flush()
+
+        observation = rdb_api.get_observation(observation_key)
+        link = rdb_api._add_observation_response_definition_link(
+            observation_id=observation.id,
+            response_definition_id=response_definition.id,
+            active_ref=active_blob.id if active_observations is not None else None,
+        )
+        for realization_number in realizations:
+            response = rdb_api.get_response(
+                name=response_key,
+                realization_index=realization_number,
+                ensemble_name=ensemble_name,
+            )
+            misfit_value = obs_vector.getTotalChi2(fs, realization_number)
+            rdb_api._add_misfit(
+                value=misfit_value, link_id=link.id, response_id=response.id
+            )
+
+
+def dump_to_new_storage(reference=None, rdb_connection=None, blob_connection=None):
+    if not FeatureToggling.is_enabled("new-storage"):
+        return
+
+    start_time = time.time()
+    print("Starting extraction...")
```

Comment:
Please search for `print`-statements and either replace them by logging or remove them...